docs/okeps/okep-5521-single-cluster-host-dpu.md

```diff
@@ -0,0 +1,308 @@
+consider a use case where we have additional interfaces on the host that we want to leverage:
+
+```mermaid
+flowchart TB
+
+  subgraph HOST[Host]
+    direction TB
+    H1["ovn-kubernetes (DPU-Host mode)"]
+    H2["ovn-kubernetes (Full mode)"]
+    H3[OVS]
+    H4[OVN]
+    H2-->H3
+    H2-->H4
+    H5[NIC]
+    H3-->H5
+  
+  end
+
+  subgraph DPU[DPU]
+    direction TB
+    D1["ovn-kubernetes (Full mode)"]
+    D2[OVS]
+    D3[OVN]
+    D4[NIC]
+    D1 --> D2
+    D1 --> D3
+    D2 --> D4
+  end
+
+  HOST <--> DPU
+```
+
+In this design, a pod may have some interfaces served via the VFs on the DPU, while other interfaces are connected via
+the OVNK running on the host. From this point forward, the OVNK running on behalf of the host will be known as "OVNK-Full Host",
+while the OVNK running on the DPU will be called "OVNK-Full DPU". The DPU-Host component will use its regular naming convention.
+
+
+## User-Stories/Use-Cases
+
+Story 1: Segment UDNs via different NICs
+
+As a cluster admin, I want certain workloads to utilize the DPU for primary UDN/cluster default network access, while
+I want to leverage my host based offloadable NICs for secondary UDNs. This will allow me to use my DPU for networks that
+have north/south capabilities, while only using my host based NICs for east/west acceleration.
+
+## Proposed Solution
+
+### Running multiple OVNK instances on the same node
+
+The first thing that must be addressed is how to actually run two OVN-Kubernetes instances that reference the same
+Kubernetes node object. OVN-Kubernetes leverages annotations on the node to indicate key information about underlay network
+and OVN configuration for the node. Some of these that are unique to the instance of OVN itself include:
+
+* k8s.ovn.org/node-encap-ips
+* k8s.ovn.org/node-chassis-id
+* k8s.ovn.org/l3-gateway-config
+
+These annotations may be different between OVNK-Full Host and OVNK-Full DPU processes. Therefore, one solution is to
+duplicate these annotations and make them specific to the DPU, for example, "dpu-encap-ips". This solution is short-sighted
+and will not accommodate for potentially in the future, using multiple DPUs. Therefore, the only other viable solution is
+to store this information in its own CR. This solution involves two potential paths:
+
+1. Use the existing node CRD and create DPU specific CRs
+2. Create a new DPUNode CRD type that mirrors the format of the Node CRD.
+
+For simplicity, it seems like going with option 1 should be the first choice, unless we hit a complication that forces us
+to use option 2.
+
+#### Creating extra node CRs for DPU instances
+
+The duplicate node CR will be created by OVNK-Full DPU instance that will realize on start up it is running with a different
+ovnk-identity. Let's assume the ovnk-identity is "dpu1-ovnk". OVNK-Full DPU will be started with a new configuration option
```

Comment:
Some similar configuration is passed via environment variables, namely `K8S_NODE_DPU` and `K8S_NODE`. How is this identity different from these variables? Can they be reused?

Reply:
I think this is used to set the node name for DPU mode to be the host node name. So in a DPU cluster, the DPU will look like another k8s node, but you want this DPU instance to serve the host, so you want to set its node name to actually be the name of the host it is serving, rather than the name of the DPU hostname. In contrast to this, you might have another OVNK instance running on the DPU that is serving the DPU itself, which would use the hostname of the DPU.

Not sure how this would interact with k8s identity yet.